namespaces/namespaces_test.go

```diff
@@ -0,0 +1,161 @@
+// +build enterprise
+
+package namespaces
+
+import (
+	"fmt"
+	"testing"
+	"time"
+
+	capi "github.com/hashicorp/consul/api"
+	"github.com/hashicorp/consul/sdk/testutil"
+	"github.com/hashicorp/consul/sdk/testutil/retry"
+	"github.com/stretchr/testify/require"
+)
+
+// Test that if the namespace already exists the function succeeds.
+func TestEnsureExists_AlreadyExists(tt *testing.T) {
+	for _, c := range []struct {
+		ACLsEnabled bool
+	}{
+		{
+			ACLsEnabled: true,
+		},
+		{
+			ACLsEnabled: false,
+		},
+	} {
+		tt.Run(fmt.Sprintf("acls: %t", c.ACLsEnabled), func(t *testing.T) {
+			req := require.New(t)
+			ns := "ns"
+
+			consul, err := testutil.NewTestServerConfigT(t, func(cfg *testutil.TestServerConfig) {
+				cfg.ACL.Enabled = c.ACLsEnabled
+				cfg.ACL.DefaultPolicy = "deny"
+			})
+			req.NoError(err)
+			defer consul.Stop()
+
+			var bootstrapToken string
+			if c.ACLsEnabled {
+				// Set up a client for bootstrapping
+				bootClient, err := capi.NewClient(&capi.Config{
```

Comment:
You didn't want to set the `masterToken` instead?

Reply:
Amazing, thank you.